### ISSUE TITLE
Compiler error in State::new()

### ISSUE THREAD

**Issue by TheBlckbird** [sotrh/learn-wgpu#545]
When trying to compile the `new` function of chapter 2 I get a compiler error.

My code should be exactly the same as in the tutorial:  
```rust
impl State {
    async fn new(window: Window) -> Self {
        let size = window.inner_size();

        let instance = Instance::new(InstanceDescriptor {
            backends: Backends::all(),
            ..Default::default()
        });

        let surface = unsafe { instance.create_surface(&window) }.unwrap();

        let adapter = instance
            .request_adapter(&RequestAdapterOptions {
                power_preference: PowerPreference::default(),
                compatible_surface: Some(&surface),
                force_fallback_adapter: false,
            })
            .await
            .unwrap();

        let (device, queue) = adapter
            .request_device(
                &DeviceDescriptor {
                    label: None,
                    required_features: wgpu::Features::empty(),
                    required_limits: if cfg!(target_arch = "wasm23") {
                        Limits::downlevel_webgl2_defaults()
                    } else {
                        Limits::downlevel_defaults()
                    },
                },
                None,
            )
            .await
            .unwrap();

        let surface_caps = surface.get_capabilities(&adapter);

        let surface_format = surface_caps
            .formats
            .iter()
            .copied()
            .filter(|f| f.is_srgb())
            .next()
            .unwrap_or(surface_caps.formats[0]);

        let config = SurfaceConfiguration {
            usage: TextureUsages::RENDER_ATTACHMENT,
            format: surface_format,
            width: size.width,
            height: size.height,
            present_mode: surface_caps.present_modes[0],
            alpha_mode: surface_caps.alpha_modes[0],
            view_formats: Vec::new(),
            desired_maximum_frame_latency: 2,
        };
        surface.configure(&device, &config);

        Self {
            surface,
            device,
            queue,
            config,
            size,
            window,
        }
    }
    ...
}
```

The error:
```
error[E0597]: `window` does not live long enough
  --> src/lib.rs:33:56
   |
25 |     async fn new(window: Window) -> Self {
   |                  ------ binding `window` declared here
...
33 |         let surface = unsafe { instance.create_surface(&window) }.unwrap();
   |                                ------------------------^^^^^^^-
   |                                |                       |
   |                                |                       borrowed value does not live long enough
   |                                argument requires that `window` is borrowed for `'static`
...
90 |     }
   |     - `window` dropped here while still borrowed

error[E0505]: cannot move out of `window` because it is borrowed
  --> src/lib.rs:88:13
   |
25 |     async fn new(window: Window) -> Self {
   |                  ------ binding `window` declared here
...
33 |         let surface = unsafe { instance.create_surface(&window) }.unwrap();
   |                                --------------------------------
   |                                |                       |
   |                                |                       borrow of `window` occurs here
   |                                argument requires that `window` is borrowed for `'static`
...
88 |             window,
   |             ^^^^^^ move out of `window` occurs here
```
   
I am using `rustc 1.77.2 (25ef9e3d8 2024-04-09)`

**Comment by TheBlckbird** [comment]
Oh, it seems like this is a proble iwth `wgpu 0.19`

**Comment by TheBlckbird** [comment]
(When) will you update the blog for `wgpu 0.19`? Anyway, thanks for your work!

**Comment by adriankiezik** [comment]
A quick fix that worked for me is creating an Arc of Window:
`let window = Arc::new(WindowBuilder::new().build(&event_loop).unwrap());`
`async fn new(window: Arc<Window>)`

Then you can pass a clone to create_surface
`let surface = instance.create_surface(window.clone()).unwrap();`
(it's a pointer to window, not a copy of window)

**Comment by sotrh** [comment]
This is fixed with merge to 0.19

**Comment by TheBlckbird** [comment]
thanks :]
